File: arrays/runningSumOf1DArrays.java
package arrays;

// https://leetcode.com/problems/running-sum-of-1d-array/
import java.util.Arrays;

public class runningSumOf1DArrays {
    public static void main(String[] args) {
        SolutionRSO1DA solution= new SolutionRSO1DA();
        System.out.println(Arrays.toString(solution.runningSum(new int [] {1,2,3,4}))); // [1,3,6,10]
        System.out.println(Arrays.toString(solution.runningSum(new int [] {1,1,1,1,1}))); // [1,2,3,4,5]

    }
    
}

class SolutionRSO1DA {
    public int[] runningSum(int[] nums) {
        int sum=0;
        
        for(int i =0; i<=nums.length-1; i++){
           nums[i] = sum+nums[i];
            sum = nums[i];
        }
        return nums;
    }
}

File: arrays/runningSumOf1DArrays.py
#https://leetcode.com/problems/running-sum-of-1d-array/

def runningSum(nums):
    #1:
    
    # summ,op=0,[]
    # for i in range(len(nums)):
    #     summ+=nums[i]
    #     op.append(summ)
    # return op
    
    #2:
    
    for i in range(1,len(nums)):
        nums[i]+=nums[i-1]
    return nums
    
    #3:
    
    # return [sum(nums[:i+1]) for i in range(len(nums))]

#input:
nums = [1,2,3,4]
print(runningSum(nums))